### ISSUE TITLE
JWT-Token is not given to Swagger-UI

### ISSUE THREAD

**Issue by timosntr** [jhipster/generator-jhipster#7444]
##### **Overview of the issue**

JWT-Token is not given to Swagger-UI.
I already experimented a bit with the SecurityConfiguration.java to get an JWT-Token in the log and test it via an REST-Client, that works. The Token that Swagger wants to work with is null.

##### **Motivation for or Use Case**

Make Swagger UI via /admin/docs usable again

##### **Reproduce the error**

1. Generate an Application
2. Login
3. Go to http://localhost:8080/#/admin/docs and try out one of the GET-Methods

##### **Related issues**

I haven't found one.

##### **Suggest a Fix**

Sorry i don't know one.

##### **JHipster Version(s)**

Release 5.0.0-beta.0

##### **JHipster configuration**

`jhipster info` output:
##### **JHipster Version(s)**

```
Release 5.0.0-beta.0
```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.pimcompany.app"
    },
    "jhipsterVersion": "5.0.0-beta.0",
    "baseName": "PIM_H2_MYSQL",
    "reactive": true,
    "packageName": "com.pimcompany.app",
    "packageFolder": "com/pimcompany/app",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "react",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Article (article) {
  name String,
  suitableManufactures String,
  oenumber Integer
}
entity MainCategory (main_category) {
  name String
}
entity SubCategory (sub_category) {
  name String
}
entity CarManufactures (car_manufactures) {
  name String
}
entity CarModel (car_model) {
  name String,
  manufacturingYear LocalDate
}
entity PartManufactures (part_manufactures) {
  name String,
  streetAddress String,
  postalCode String,
  city String,
  stateProvince String
}
relationship OneToMany {
  SubCategory{artciles} to Article{subCategory},
  CarModel{suitableArticles} to Article{carModel},
  PartManufactures{items} to Article{partManufactures},
  MainCategory{subCategories} to SubCategory{mainCategory},
  CarManufactures{models} to CarModel{carManufactures}
}

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)

git version 2.15.0.windows.1

node: v9.1.0

npm: 5.5.1

yeoman: 2.0.1

yarn: 1.3.2

Docker version 17.09.0-ce, build afdb6d4

docker-compose version 1.16.1, build 6d1ac219

##### **Environment and Tools**

java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)

git version 2.15.0.windows.1

node: v9.1.0

npm: 5.5.1

yeoman: 2.0.1

yarn: 1.3.2

Docker version 17.09.0-ce, build afdb6d4

docker-compose version 1.16.1, build 6d1ac219

##### **Browsers and Operating System**

Windows
Tested on Chrome and Firefox

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by jdubois** [comment]
This would be really surprising - I'm testing right now!

**Comment by timosntr** [comment]
Alright! Thank you very much! In older versions it was always working without BrowserSync, so i didn't use it in 5.0.0beta as an habit.

**Comment by jdubois** [comment]
Yes I don't know what happened here - I guess it's just because we don't use port 8080, and hopefully that's just a simple configuration trick

**Comment by timosntr** [comment]
Maybe you did something different, but on BrowserSync it isn't working for me neither.

**Comment by jdubois** [comment]
I confirm it works with BrowserSync, I have it right in front of me. Are you sure you connect through port 9000 and are logged in? Can you see the 'Authorization: Bearer XXXX' header in the CURL example?

**Comment by timosntr** [comment]
I can ensure you that I'm connected through port 9000 and I'm logged in. My header in the CURL example is -like on port 8080- 'Authorization: Bearer null'

**Comment by jdubois** [comment]
That's strange, I only get this on port 8080 - which again, is not normal. I've just done some quick tests, I need to look at the code to understand what's going on, but I don't have the time right now.

**Comment by deepu105** [comment]
Are you on HTTPS?

Thanks & Regards,
Deepu

**Comment by timosntr** [comment]
Hey Deepu,
No I‘m using just HTTP.

**Comment by pyotrrtichie** [comment]
I was experiencing similar issues to @timosntr when using browsersync where the token was null. I created a new gateway to check if I had messed something up and using browser sync i get "Can't read swagger JSON from http://localhost:9000/yds/v2/api-docs". But if I use the server on 8080 I have no issues. Any thoughts on why this might be?

Without using browsersync on my gateway this similarly 'solves' the problem. If I go back to using browsersync the json for swagger is served but I get a 403 and "Could not verify the provided CSRF token because your session was not found".

**Comment by eruixma** [comment]
I might find the root cause for this issue, it is because a typo in `webapp/swagger-ui/index.html`
`function addApiKeyAuthorization() {
                    var authToken = JSON.parse(localStorage.getItem("jhi-authenticationtoken") || sessionStorage.getItem("jhi-authenticationtoken"));
                    var apiKeyAuth = new SwaggerClient.ApiKeyAuthorization("Authorization", "Bearer " + authToken, "header");
                    window.swaggerUi.api.clientAuthorizations.add("bearer", apiKeyAuth);
                }`

the key `jhi-authenticationtoken` should be `jhi-authenticationToken`

**Comment by deepu105** [comment]
Strange it was working fine, as far as I remember anyway can you do a PR to
fix
